src/auth/oauth2client.ts

```diff
@@ -296,6 +296,12 @@ export interface GenerateAuthUrlOpts {
    * must be used with the 'code_challenge' parameter described above.
    */
   code_challenge?: string;
+
+  /**
+   * A way for developers and/or the auth team to provide a set of key value
+   * pairs to be added as query parameters to the authorization url.
+   */
+  [key: string]: querystring.ParsedUrlQueryInput;
```

Comment:
Apologies, it looks like it's a bit difficult to get to the root types. I tried this locally and it works:

```suggestion
  [
    key: string
  ]: querystring.ParsedUrlQueryInput[keyof querystring.ParsedUrlQueryInput]
```

Reply:
Thank you for checking locally! I didn't even see that it had failed!